### DISCUSSION TITLE
->defaultItems(1) on Repeater did not show empty row

### DISCUSSION THREAD

**Discussion by ziming** [filamentphp/filament#2851] [Help]
### Package

filament/forms

### Package Version

2.13.7

### Laravel Version

9

### Livewire Version

2

### PHP Version

8.1

### Bug description

Repeater::make('Chapters')
->relationship('chapters')
->formSchema([...])
->defaultItems(1);

I expect 1 empty row to be shown by default. Instead I still have to press the add button to show it.

**Comment by danharrin** [comment]
Make sure you call `$this->form->fill()` in `mount()`

**Reply by ziming** [reply]
Yes I did call that

**Reply by danharrin** [reply]
Can you post the entire form code?

**Reply by ziming** [reply]
my entire form code is tons of lines and had some sensitive stuffs.

but I can say my repeater is in a card and my card is in a form wizard step

**Reply by danharrin** [reply]
I cannot help you if you don't post code I can try. I cannot replicate this myself.

**Reply by ziming** [reply]
ok I will post a subset of the form code when I can

**Reply by xavvey** [reply]
It also doesn't work in my code. I am using the Repeater to create a new user and link hasMany relationship to it, so I am not using the `$this->form->fill()` method because I don't have any data to fill the form with (except with these children)

Here is my code for the repeater

    ```
            Step::make('Add Child')
                    ->schema([
                        Repeater::make('children')
                            ->schema([
                                Select::make('children')
                                    ->options(
                                        Service::query()
                                            ->select('name')
                                            ->orderBy('name')
                                            ->get()
                                            ->keyBy('name')
                                            ->map(fn($child) => $child->name)
                                            ->toArray()
                                    )
                           ])
                            ->createItemButtonLabel('Add Child')
                            ->defaultItems(1),
                    ]),
```

setting `defaultItems()` to any amount does nothing..
Any thought?

**Reply by danharrin** [reply]
You need to call `$this->form->fill()`, as that fills the defaults, if you have no data. It is mandatory and will fix your issue.

**Reply by xavvey** [reply]
@danharrin Thanks for the fast reply. I must have misunderstood the documentation because I thought it only needed to be used if there was already data to use in a form (like for an Edit page or something)

This fixed it indeed, so thanks!
